packages/zealot/src/query/result-stream.ts

```diff
@@ -0,0 +1,118 @@
+import {isObject} from "lodash"
+import {Response as NodeResponse} from "node-fetch"
+import {zjson} from ".."
+import {eachLine} from "../ndjson/lines"
+import {JSOptions} from "../zed/values/types"
+import {Channel, Collector} from "./channel"
+
+type CrossResponse = Response | NodeResponse
+export class ResultStream {
```

Comment:
This is the class that handles the stream of zed values coming back from the server when you issue a query. 

```
const res = await client.query("hello")
await res.js()
await res.collect(({rows, shapes}) => {
  setRows(rows)
})
```
The above example is for a single channel result.

If you have multiple channels, all the methods are the same, you just get the channel first.

```
await res.channel(3).collect(() => {})
await res.channel(2).js()
await res.channel(5).shapes
/// and so on
```

The `.promise` method returns a promise for when the whole thing is done.

Reply:
This looks way cleaner to me now. Though I still wish the channel system could use strings instead of numbers to name and refer to each channel. Not sure how much we are in control of that though with the behavior of backend